.github/workflows/migrations.yml

```diff
@@ -1,18 +1,49 @@
 name: migrations
-on:
-  pull_request:
-    paths:
-      - 'src/sentry/migrations/*'
+on: pull_request
 
 jobs:
+  did-migration-change:
+    name: check if any migration changes
+    runs-on: ubuntu-20.04
+    timeout-minutes: 3
+    # Map a step output to a job output
+    outputs:
+      added: ${{ steps.changes.outputs.migrations_added }}
+      modified: ${{ steps.changes.outputs.migrations_modified }}
+    steps:
+      - name: Checkout sentry
+        uses: actions/checkout@v2
+
+      - name: Match migration files
+        uses: getsentry/paths-filter@v2
+        id: changes
+        with:
+          token: ${{ github.token }}
+          filters: .github/file-filters.yml
+
+  modified-migration:
+    name: check if modified migration
+    runs-on: ubuntu-20.04
+    timeout-minutes: 4
+    needs: did-migration-change
+    if: ${{ needs.did-migration-change.outputs.modified == 'true' }}
+
+    steps:
+      - name: Failure because of modified migration
+        shell: bash
+        run: |
+          echo "We do not allow modifying migrations. A GH admin can still merge this." && exit 1
+
   sql:
     name: Generate SQL
     runs-on: ubuntu-20.04
     timeout-minutes: 8
+    needs: did-migration-change
+    if: ${{ needs.did-migration-change.outputs.added == 'true' }}
 
     steps:
       # Checkout master to run all merged migrations.
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
```

Comment:
FYI, this is a breaking change between `v1` and `v2` - the former fetches ALL refs by default and `v2` only fetches a single commit. This may affect the `Get changed migration files` step.

However we could change that step to use the output of `steps.changes.outputs.migrations_added_files`

Reply:
Moving this to #24143